CONTRIBUTING.md

```diff
@@ -1,43 +1,86 @@
+Reporting bugs is one of the best ways to contribute. However, a good bug report has some very specific qualities, so please read over our short document on [reporting bugs](./doc/dev/reporting_bugs.md) before submitting a bug report. Before filing a bug report, ensure the bug hasn't already been reported by searching through the OLM project [Issues][issues].
+
+Any new contribution should be accompanied by a new or existing issue. This issue can help track work, discuss the design and implementation, and help avoid wasted efforts or multiple people working on the same issue, compared to submitting a PR first. Trivial changes, like fixing a typo in the documentation, do not require the creation of a new issue.
+
+Proposing larger changes to the OLM project may require an enhancement be created in the [operator-framework/enhancements](https://github.com/operator-framework/enhancements/) repository. Enhancements are the primary mechanism for proposing new features to the OLM codebase. Any change to OLM's behavior or existing features, APIs, or architectural changes to the testing harness likely require an enhancement.
 
 ## Contribution flow
 
 This is a rough outline of what a contributor's workflow looks like:
 
-- Create a topic branch from where to base the contribution. This is usually master.
+- Identify or create an issue.
+- Create a topic branch from where to base the contribution. This is usually the master branch.
 - Make commits of logical units.
 - Make sure commit messages are in the proper format (see below).
+- Ensure all relevant commit messages contain a valid sign-off message (see below).
 - Push changes in a topic branch to a personal fork of the repository.
-- Submit a pull request to operator-framework/operator-lifecycle-manager.
-- The PR must receive a LGTM from two maintainers found in the MAINTAINERS file.
+- Submit a pull request to the operator-framework/operator-lifecycle-manager repository.
+- Wait and respond to feedback from the maintainers listed in the OWNERS file.
 
 Thanks for contributing!
 
+### Code Review
+
+Contributing PRs with a reasonable title and description can go a long way with helping ease the burden of the review process.
+
+It can be helpful after submitting a PR to self-review the changes yourself. This allows you to communicate sections that reviewers should spend time combing over, asking for feedback on a particular implementation, or providing justification for a set of changes in-line ahead of time.
+
+When opening PRs that are in a rough draft or WIP state, prefix the PR description with `WIP: ...` or create a draft PR. This can help save reviewer's time communicating the state of a PR ahead of time. Draft/WIP PRs can be a good way to get early feedback from reviewers on the implementation, focusing less on smaller details, and more on the general approach of changes.
```

Comment:
```suggestion
When opening PRs that are in a rough draft or WIP state, prefix the PR description with `WIP: ...` or create a draft PR. This can help save reviewer's time by communicating the state of a PR ahead of time. Draft/WIP PRs can be a good way to get early feedback from reviewers on the implementation, focusing less on smaller details, and more on the general approach of changes.
```